crypto/x509/x509_vfy.c

```diff
@@ -494,23 +494,27 @@ static int check_chain(X509_STORE_CTX *ctx)
                          && ret != 1), ctx, x, i, X509_V_ERR_INVALID_CA);
             break;
         }
-        if (num > 1) {
-            /* Check for presence of explicit elliptic curve parameters */
-            ret = check_curve(x);
-            CHECK_CB(ret < 0, ctx, x, i, X509_V_ERR_UNSPECIFIED);
-            CHECK_CB(ret == 0, ctx, x, i, X509_V_ERR_EC_KEY_EXPLICIT_PARAMS);
-        }
+
         /*
-         * Do the following set of checks only if strict checking is requested
-         * and not for self-issued (including self-signed) EE (non-CA) certs
+         * Skip further checks for self-issued (including self-signed) EE certs
          * because RFC 5280 does not apply to them according RFC 6818 section 2.
          */
-        if ((ctx->param->flags & X509_V_FLAG_X509_STRICT) != 0
-            && num > 1) { /*
-                           * this should imply
-                           * !(i == 0 && (x->ex_flags & EXFLAG_CA) == 0
-                           *          && (x->ex_flags & EXFLAG_SI) != 0)
-                           */
+        if (i == 0 && (x->ex_flags & EXFLAG_CA) == 0
+                && (x->ex_flags & EXFLAG_SI) != 0)
```

Comment:
I am not sure this is correct improvement. This will start rejecting CA certificates used as self signed EE certs with the strict checks.

Reply:
Yeah, that check was wrongly stated. 
It was meant to detect (and then exclude) directly trusted self-issued EE certs.
To detect directly trusted certs, `i == 0` is wrong, but `num == 1` can be used.
Corrected.